management_app/views/course.py

```diff
@@ -2,22 +2,140 @@
+        #         ' VALUES (?, ?, ?, ?, ?)',
+        #         (course_title_id, course_title, units, teaching_point_val, course_level)
+        #     )
+        #     db.commit()
+
+        # TODO: haven't calculate "teaching_point_val" column
+
+        return render_template('courses/catalog.html')
+
+
+@courses.route('/data-templates/<filename>', methods=['GET'])
+@login_required
+def download_template(filename):
+
+    # TODO: do template Prefill: 
+    # Schedule_teaching table prepopulate: year, quarter, usename, ucinetid, 
+    # course_type： Lec, course_sec: A (each prodessor 3 rows: fall, winter, spring)
+
+    if request.method == 'GET':
+        return download_file(filename)
+
+@courses.route('/upload', methods=['POST'])
+@login_required
+def upload_user_file():
+    if (request.method == 'POST'):
+        file = request.files['courseTemplate'] 
+        upload_file(file)        
+        quarterDict = {"Fall": 1, "Winter": 2, "Spring": 3, "Summer": 4}
+        db = get_db()
+
+        df = pd.read_excel(get_upload_filepath(file.filename), sheet_name=1)
+        rows = df.values.tolist()   
+        for row in rows:
+            year = row[0]
+            quarter = row[1]
+            if quarter in quarterDict.keys():
+                quarter = quarterDict[quarter]
+            user_UCINetID = row[2].strip()
+            course_code = row[3]
+
+            # course_title_id column may be like "CS 143B" or "CS143B" -> all convert to "CS143B"
+            course_title_id = row[4].strip().replace(' ', '')
+
+            course_type = row[5].strip()
+            # course_type = "Lec"
+            course_sec = row[6].strip()
+            # course_sec = 'A'
+            num_of_enrollment = row[7]
+            offload_or_recall_flag = row[8]
+            
+            user_id = None
+            # use "user_UCINetID" in s_t file to get user_id in users table
+            # take ucinetid: alfchen, emj to test
+            row = db.execute(
+                'SELECT user_id FROM users'
+                ' WHERE user_ucinetid = ?', (user_UCINetID,)
+            ).fetchone()
+            if row != None:
+                user_id = row[0]
+            
+            course_id = None
+            # use "course_title_id" in s_t file to get course_id in courses table
+            row = db.execute(
+                'SELECT course_id FROM courses'
+                ' WHERE course_title_id = ?', (course_title_id,)
+            ).fetchone()
+            if row != None:
+                course_id = row[0]
+
+            db.execute(
+                'INSERT INTO scheduled_teaching (user_id, course_code, year, quarter, course_id, course_type, course_sec, enrollment, offload_or_recall_flag)'
+                ' VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)',
+                (user_id, course_code, year, quarter, course_id, course_type, course_sec, num_of_enrollment, offload_or_recall_flag)
+            )
```

Comment:
Because there is only "insert" SQL to create new rows in the database scheduled_teaching table. Do we have edit scenarios, which will have "update" SQL, for uploading the scheduled_teaching template?

Reply:
will do this in the next commit